src/boot/efi/log.c

```diff
@@ -7,16 +7,48 @@
 
 static unsigned log_count = 0;
 
+#define DEBUGCON_PREFIX u"systemd: "
+#define DEBUGCON_ACK 0xE9
+#define DEBUGCON_PORT 0x402
+
 void freeze(void) {
         for (;;)
                 BS->Stall(60 * 1000 * 1000);
 }
 
+#if defined(__i386__) || defined(__x86_64__)
+static bool log_has_debugcon(void) {
+        static bool init;
+        static bool present;
```

Comment:
hmm, we would usually do that with a "tri-state", i.e.

```
static int present = -1;

if (present < 0)
        present = inb(…

return present;
```
```

Reply:
Yep, already fixed locally